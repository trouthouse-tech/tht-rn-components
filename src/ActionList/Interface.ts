export interface ActionListItemProps {
    title: string;
    route: string;
}

export interface ActionListProps {
    items: ActionListItemProps[];
}
