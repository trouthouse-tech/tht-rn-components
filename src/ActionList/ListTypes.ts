export type ActionListItemProps = {
  title: string;
  route: string;
};

export type ActionListProps = {
  items: ActionListItemProps[];
};
