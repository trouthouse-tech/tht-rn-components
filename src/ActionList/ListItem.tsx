import React from 'react';

import { View, TouchableOpacity, StyleSheet, Text, Image } from 'react-native';

import { ActionListItemProps } from './ListTypes';
import { Colors } from '../styles/Color';

const ListItem: React.FC<ActionListItemProps> = ({ title, route }) => {
    const onPress = () => {
        console.log(route);
    };

    return (
        <View style={styles.listItemStyle}>
            <TouchableOpacity onPress={onPress} style={styles.touchStyle}>
                <View>
                    <Text style={styles.textStyle}>{title}</Text>
                </View>
                <Image
                    width={20}
                    height={20}
                    source={require('./img/rightArrow.png')}
                />
            </TouchableOpacity>
        </View>
    );
};

export default ListItem;

const styles = StyleSheet.create({
    listItemStyle: {
        borderBottomWidth: StyleSheet.hairlineWidth,
        padding: 20,
        paddingVertical: 20,
        borderRadius: 15,
    },
    touchStyle: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignContent: 'center',
    },
    textStyle: {
        fontSize: 18,
        fontWeight: '600',
        color: Colors.Black,
    },
});
