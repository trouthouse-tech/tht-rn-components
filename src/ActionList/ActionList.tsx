import React from 'react';

import { View, FlatList, StyleSheet } from 'react-native';

import { ActionListProps, ActionListItemProps } from './Interface';
import ListItem from './ListItem';

const ActionList: React.FC<ActionListProps> = ({ items }) => {
    return (
        <View style={styles.listStyle}>
            <FlatList
                keyExtractor={(item: ActionListItemProps, index: number) =>
                    index.toString()
                }
                data={items}
                renderItem={({ item }: { item: ActionListItemProps }) => (
                    <ListItem title={item.title} route={item.route} />
                )}
            />
        </View>
    );
};

export default ActionList;

const styles = StyleSheet.create({
    listStyle: {
        borderWidth: StyleSheet.hairlineWidth,
        borderRadius: 15,
    },
});
