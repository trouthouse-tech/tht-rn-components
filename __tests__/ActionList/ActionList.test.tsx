import { render, cleanup } from 'react-native-testing-library';
import React from 'react';

import ActionList from '../../src/ActionList/ActionList';
import { ActionListProps } from '../../src/ActionList/ListTypes';

afterEach(cleanup);

describe('<ActionList />', () => {
    const actions: ActionListProps = {
        items: [
            {
                title: 'Edit Account',
                route: 'EditAccount',
            },
            {
                title: 'Terms and Conditions',
                route: 'TermsAndConditions',
            },
        ],
    };

    it('should match snapshot', () => {
        const rendered = render(<ActionList items={actions.items} />).toJSON();

        expect(rendered).toMatchSnapshot();
    });
});
