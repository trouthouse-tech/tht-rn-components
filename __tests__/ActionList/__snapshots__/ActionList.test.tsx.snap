// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionList /> should match snapshot 1`] = `
<View
  style={
    Object {
      "borderRadius": 15,
      "borderWidth": 0.5,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "route": "EditAccount",
          "title": "Edit Account",
        },
        Object {
          "route": "TermsAndConditions",
          "title": "Terms and Conditions",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderBottomWidth": 0.5,
              "borderRadius": 15,
              "padding": 20,
              "paddingVertical": 20,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignContent": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "fontWeight": "600",
                  }
                }
              >
                Edit Account
              </Text>
            </View>
            <Image
              height={20}
              source={
                Object {
                  "testUri": "../../../src/ActionList/img/rightArrow.png",
                }
              }
              width={20}
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderBottomWidth": 0.5,
              "borderRadius": 15,
              "padding": 20,
              "paddingVertical": 20,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignContent": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 18,
                    "fontWeight": "600",
                  }
                }
              >
                Terms and Conditions
              </Text>
            </View>
            <Image
              height={20}
              source={
                Object {
                  "testUri": "../../../src/ActionList/img/rightArrow.png",
                }
              }
              width={20}
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
